src/AWS.Deploy.CLI/Commands/CommandFactory.cs

```diff
@@ -307,6 +309,72 @@ private Command BuildListCommand()
             return listCommand;
         }
 
+        private Command BuildDeploymentProjectCommand()
+        {
+            var deploymentProjectCommand = new Command("deployment-project",
+                "Save the CDK deployment project inside a user provided directory path.");
+
+            var generateDeploymentProjectCommand = new Command("generate",
+                "Save the CDK deployment project inside a user provided directory path without proceeding with a deployment")
+            {
+                _optionOutputDirectory,
+                _optionDiagnosticLogging,
+                _optionProjectPath
+            };
+
+            generateDeploymentProjectCommand.Handler = CommandHandler.Create(async (GenerateDeploymentProjectCommandHandlerInput input) =>
+            {
+                try
+                {
+                    _toolInteractiveService.Diagnostics = input.Diagnostics;
+                    var projectDefinition = await _projectParserUtility.Parse(input.ProjectPath ?? "");
+
+                    var saveDirectory = input.Output ?? "";
+
+                    OrchestratorSession session = new OrchestratorSession(projectDefinition, "", "");
+
+                    var generateDeploymentProject = new GenerateDeploymentProjectCommand(
+                        _toolInteractiveService,
+                        _consoleUtilities,
+                        _orchestratorInteractiveService,
+                        _cdkProjectHandler,
+                        _cdkManager,
+                        _awsResourceQueryer,
```

Comment:
We should break the Orchestrator. `IAWSResourceQueryer`, `ICDKManager`, `IDeploymentBundleHandler` are not required to generate the deployment project.

Reply:
Added an `Orchestrator` constructor that only takes the `OrchestratorSession` and `RecipeDefinitionPaths`